registry/manifest.go

```diff
@@ -116,7 +116,7 @@ func (registry *Registry) PutManifest(repository, reference string, signedManife
 		return err
 	}
 
-	req.Header.Set("Content-Type", schema1.MediaTypeManifest)
```

Comment:
This only worked because the schema 1 registers as an empty string (https://github.com/docker/distribution/blob/7484e51bf6af0d3b1a849644cdaced3cfcf13617/manifest/schema1/manifest.go#L50) and that's the fallback (https://github.com/docker/distribution/blob/7484e51bf6af0d3b1a849644cdaced3cfcf13617/manifests.go#L108) if you provide an unknown media type. (That's really bad design. An unknown value should be an error.)